testutil/capturelog.go

```diff
@@ -4,79 +4,129 @@ import (
 	"fmt"
 	"log"
 	"sync"
+	"testing"
 
 	"github.com/influxdata/telegraf"
 )
 
 var _ telegraf.Logger = &CaptureLogger{}
 
+const (
+	logLevelDebug = 'D'
+	logLevelWarn  = 'W'
+	logLevelInfo  = 'I'
+	logLevelError = 'E'
+)
+
+type logMessage struct {
+	level   byte
+	message string
+}
+
 // CaptureLogger defines a logging structure for plugins.
 type CaptureLogger struct {
-	Name   string // Name is the plugin name, will be printed in the `[]`.
-	errors []string
+	T        *testing.T
+	Name     string // Name is the plugin name, will be printed in the `[]`.
+	messages []logMessage
 	sync.Mutex
 }
 
-// Errorf logs an error message, patterned after log.Printf.
-func (l *CaptureLogger) Errorf(format string, args ...interface{}) {
-	s := fmt.Sprintf("E! ["+l.Name+"] "+format, args...)
+func (l *CaptureLogger) msgString(msg logMessage) string {
+	return fmt.Sprintf("%c! [%s] %s", msg.level, l.Name, msg.message)
+}
+func (l *CaptureLogger) logMsg(msg logMessage) {
 	l.Lock()
-	l.errors = append(l.errors, s)
+	l.messages = append(l.messages, msg)
+	l.T.Log(l.msgString(msg))
 	l.Unlock()
-	log.Print(s)
+}
+
+func (l *CaptureLogger) logf(level byte, format string, args ...any) {
+	l.logMsg(logMessage{level, fmt.Sprintf(format, args...)})
+}
+
+func (l *CaptureLogger) loga(level byte, args ...any) {
+	l.logMsg(logMessage{level, fmt.Sprint(args...)})
+}
```

Comment:
I really like this! We should also do this in the actual logger, but that's for another PR...